types/simulation/types.go

```diff
@@ -95,11 +101,36 @@ func NewOperationMsg(msg sdk.Msg, ok bool, comment string, cdc *codec.ProtoCodec
 		moduleName = msgType
 	}
 
-	if legacyMsg, okType := msg.(legacytx.LegacyMsg); okType {
```

Comment:
Another option I explored was packing sdk.Msg into an any message and then marshaling that, but if we do
the encoder will reject messages that are not tagged with an amino.name.  Strictly speaking we don't need
an amino.name on messages which aren't expected to be packed into any fields except for this one case.